flexbox/src/main/java/com/google/android/flexbox/FlexboxLayoutManager.java

```diff
@@ -2873,17 +2882,38 @@ private void assignFromView(View anchor) {
                 // We need to use the anchor view as starting from right if the flex direction is
                 // (column or column_reverse) and layout direction is RTL.
                 if (mLayoutFromEnd) {
-                    mCoordinate = mOrientationHelper.getDecoratedStart(anchor) +
-                            mOrientationHelper.getTotalSpaceChange();
+                    // When the flex wrap is nowrap, the orientation helper is changed to
```

Comment:
Why not add something like this at the beginning of the method:
```java
OrientationHelper orientationHelper;
// When the flex wrap is nowrap, the orientation helper is changed to
// perpendicular. Need to use the mSubOrientationHelper to be consistent with
// when flex wrap is not nowrap.
if (mFlexWrap == FlexWrap.NOWRAP) {
    orientationHelper = mSubOrientationHelper;
} else {
    orientationHelper = mOrientationHelper;
}
```

And then using directly `orientationHelper` in the four changes you made in this method?

Reply:
It looks better. Thanks.